src/plugins/intel_cpu/src/cpu_tensor.cpp

```diff
@@ -0,0 +1,101 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <memory>
+#include <string>
+#include <map>
+
+#include "cpu_tensor.h"
+#include "ie_blob.h"
+#include "ie_ngraph_utils.hpp"
+#include "openvino/runtime/iremote_tensor.hpp"
+#include "openvino/runtime/properties.hpp"
+
+namespace ov {
+namespace intel_cpu {
+
+//For the beginning probably one contructor should be enough
+Tensor::Tensor(MemoryPtr memptr) : m_memptr{memptr} {
+    OPENVINO_ASSERT(m_memptr != nullptr);
+
+    auto& desc = m_memptr->getDesc();
```

Comment:
Suggest inserting the `ncsp` check here and remove `get_strides()` call, thus we update the strides only on demand, (i.e. when the user calls get_strides())

Reply:
done.